### PR TITLE
Beautify code (implements #693)

### PR DESCRIPTION
**Pull request by skorokithakis**
I have added pre-commit to CI and added `black/reorder-python-imports/flake8` to it. This should help keep the code style consistent and weed out many bugs before even committing.

### CODE REVIEW

**Comment by nixjdm** [comment]
Ref - Fixes #693

**Comment by skorokithakis** [comment]
Oh huh, I thought the title reference would work, thanks.

**Comment by nixjdm** [comment]
All maintainers who see this, wait on the merge until after 3.2 is cut please. :) I want to do that this month.

**Comment by skorokithakis** [comment]
Fine, fine, I won't merge yet :P

**Comment by runfalk** [comment]
I think we should also update the *Want to develop on Lektor?* section in the README on how to set up a pre-commit. Alternatively we should add support to the `make install-git-hooks` command.

I also think this is a good time to change the README to be Python 3 favored (use `python3 -m venv --prompt="(lektor)" venv` rather than `virtualenv venv`).

Maybe we should also add `black` to `setup.py` as an `extra_requires` target (named `dev`)?

**Comment by skorokithakis** [comment]
I agree with everything @runfalk says, except the additional dependency, since pre-commit installs its tools itself in a virtualenv.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+repos:
+-   repo: https://github.com/ambv/black
+    rev: 18.9b0
```
Comment (nixjdm) [inline .pre-commit-config.yaml]: These versions should be updated and the tools rerun before merge to keep out as much churn as we can.

Comment (davidism) [reply .pre-commit-config.yaml]: `pre-commit autoupdate` will set the latest versions, then `pre-commit run --all-files` will reapply them.

**Comment by nixjdm** [comment]
Thank you @runfalk for [mentioning git hyper merge](https://github.com/lektor/lektor/issues/693#issuecomment-538289403). I hadn't heard of it. It sounds good to me to add a short blurb about it to the readme. At the least, adding a `.git-blame-ignore-revs` seems like a good move. To support that most, all of the formatting changes should be in a single commit (or at least black). For instance, this whole PR could be squashed into a single commit, and one additional that adds `.git-blame-ignore-revs` with the previous commit's hash.

The readme could also use a black badge.

Sound good @skorokithakis?

**Comment by skorokithakis** [comment]
Sure thing, done.

**Comment by skorokithakis** [comment]
What's the status of this? I imagine it has gotten pretty stale.

**Comment by runfalk** [comment]
I guess we need to release 3.2 first, which is blocked by #698

**Comment by yagebu** [comment]
@skorokithakis: 3.2 is out, so we can do this now :) Do you have time to get this up-to-date or do you want me to take this over?

**Comment by skorokithakis** [comment]
It's fine @yagebu, I can do it, though we should do the following:

* Scrap this branch.
* Branch again from master.
* Add the configs and beautify everything with them.
* Add the commit SHA to the ignore file.
* Merge.

**Comment by skorokithakis** [comment]
Closing in favor of #823.

**Pull request closed without merging**